<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CKI_DGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Setting Up
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="obtainData.html">Obtaining Data</a>
    </li>
    <li>
      <a href="setUp.html">QC and Pre-processing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge.html">DE Analysis</a>
    </li>
    <li>
      <a href="go.html">GO Term Enrichment Analysis</a>
    </li>
    <li>
      <a href="kegg.html">KEGG Pathway Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>01_CKI_DGE/1_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha13501CKIDGEtree848e971e9ccbf102898eeb089076953d68a0817ftargetblank848e971a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tranmanhha135/01_CKI_DGE/tree/848e971e9ccbf102898eeb089076953d68a0817f" target="_blank">848e971</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha13501CKIDGEtree848e971e9ccbf102898eeb089076953d68a0817ftargetblank848e971a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tranmanhha135/01_CKI_DGE/tree/848e971e9ccbf102898eeb089076953d68a0817f" target="_blank">848e971</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  0_data/rds_object/qlf_all.rds
    Untracked:  0_data/rds_object/qlf_sig.rds
    Untracked:  0_data/rds_object/treat_all.rds
    Untracked:  0_data/rds_object/treat_sig.rds
    Untracked:  1_analysis/about.html
    Untracked:  1_analysis/figure/dge.Rmd/NBDispersion-1.png
    Untracked:  1_analysis/figure/dge.Rmd/coolmap-1.png
    Untracked:  1_analysis/figure/dge.Rmd/heatmap_common-1.png
    Untracked:  1_analysis/figure/dge.Rmd/heatmap_unique-1.png
    Untracked:  1_analysis/figure/dge.Rmd/qlf_dispersion-1.png
    Untracked:  1_analysis/figure/dge.Rmd/volcano_treat-1.png
    Untracked:  1_analysis/figure/dge.Rmd/volcano_treat-2.png
    Untracked:  1_analysis/figure/dge.Rmd/volcano_treat-3.png
    Untracked:  1_analysis/figure/dge.Rmd/volcano_treat-4.png
    Untracked:  1_analysis/figure/dge.Rmd/volcano_treat-5.png
    Untracked:  1_analysis/figure/go.Rmd/
    Untracked:  1_analysis/figure/kegg.Rmd/completePathway-1.png
    Untracked:  1_analysis/figure/kegg.Rmd/dotPlot-1.png
    Untracked:  1_analysis/figure/kegg.Rmd/partialPathway, -1.png
    Untracked:  1_analysis/figure/kegg.Rmd/partialPathway-1.png
    Untracked:  1_analysis/figure/setUp.Rmd/afterFiltering-2.png
    Untracked:  2_plots/1_setup/after_normalisation.svg
    Untracked:  2_plots/2_dge/MA_treat_CKI-Mac vs CKI.svg
    Untracked:  2_plots/2_dge/MA_treat_CKI-Nme vs CKI.svg
    Untracked:  2_plots/2_dge/MA_treat_CKI-Omt vs CKI.svg
    Untracked:  2_plots/2_dge/MA_treat_CKI-Tri vs CKI.svg
    Untracked:  2_plots/2_dge/MA_treat_UT vs CKI.svg
    Untracked:  2_plots/2_dge/NB_dispersion.svg
    Untracked:  2_plots/2_dge/heatmap_logCPM_common.svg
    Untracked:  2_plots/2_dge/heatmap_logCPM_unique.svg
    Untracked:  2_plots/2_dge/pValue_treat_CKI-Mac vs CKI.svg
    Untracked:  2_plots/2_dge/pValue_treat_CKI-Nme vs CKI.svg
    Untracked:  2_plots/2_dge/pValue_treat_CKI-Omt vs CKI.svg
    Untracked:  2_plots/2_dge/pValue_treat_CKI-Tri vs CKI.svg
    Untracked:  2_plots/2_dge/pValue_treat_UT vs CKI.svg
    Untracked:  2_plots/2_dge/volcano_treat_CKI-Mac vs CKI.svg
    Untracked:  2_plots/2_dge/volcano_treat_CKI-Nme vs CKI.svg
    Untracked:  2_plots/2_dge/volcano_treat_CKI-Omt vs CKI.svg
    Untracked:  2_plots/2_dge/volcano_treat_CKI-Tri vs CKI.svg
    Untracked:  2_plots/2_dge/volcano_treat_UT vs CKI.svg
    Untracked:  2_plots/3_go/
    Untracked:  2_plots/4_kegg/
    Untracked:  3_output/enrichKEGG_treat_sig.xlsx
    Untracked:  3_output/goana_treat.xlsx
    Untracked:  3_output/kegga_treat_sig.xlsx
    Untracked:  3_output/qlf_decideTest.xlsx
    Untracked:  3_output/qlf_sig.xlsx
    Untracked:  3_output/treat_decideTest.xlsx
    Untracked:  3_output/treat_sig.xlsx
    Untracked:  Meeting Notes.pdf

Unstaged changes:
    Modified:   0_data/rds_object/dge.rds
    Deleted:    0_data/rds_object/dge_treat_unfiltered.rds
    Deleted:    0_data/rds_object/enrichGO_treat.rds
    Deleted:    0_data/rds_object/enrichKEGG_treat_filtered.rds
    Modified:   0_data/rds_object/qlf.rds
    Deleted:    0_data/rds_object/qlf_filtered.rds
    Modified:   0_data/rds_object/treat.rds
    Deleted:    0_data/rds_object/treat_filtered.rds
    Deleted:    0_data/rds_object/treat_unfiltered.rds
    Modified:   1_analysis/dge.Rmd
    Modified:   1_analysis/figure/dge.Rmd/MAplotCustom_treat-1.png
    Modified:   1_analysis/figure/dge.Rmd/MAplotCustom_treat-2.png
    Modified:   1_analysis/figure/dge.Rmd/MAplotCustom_treat-3.png
    Modified:   1_analysis/figure/dge.Rmd/MAplotCustom_treat-4.png
    Modified:   1_analysis/figure/dge.Rmd/MAplotCustom_treat-5.png
    Modified:   1_analysis/figure/dge.Rmd/pValueHistogram_treat-1.png
    Modified:   1_analysis/figure/dge.Rmd/pValueHistogram_treat-2.png
    Modified:   1_analysis/figure/dge.Rmd/pValueHistogram_treat-3.png
    Modified:   1_analysis/figure/dge.Rmd/pValueHistogram_treat-4.png
    Modified:   1_analysis/figure/dge.Rmd/pValueHistogram_treat-5.png
    Modified:   1_analysis/figure/setUp.Rmd/MDSplot-1.png
    Modified:   1_analysis/figure/setUp.Rmd/afterFiltering-1.png
    Modified:   1_analysis/figure/setUp.Rmd/libraryChecks-1.png
    Modified:   1_analysis/figure/setUp.Rmd/pcaSampleGroup-1.png
    Modified:   1_analysis/figure/setUp.Rmd/pcaSampleType-1.png
    Modified:   1_analysis/figure/setUp.Rmd/visualisingNormalisation-1.png
    Modified:   1_analysis/go.Rmd
    Modified:   1_analysis/kegg.Rmd
    Modified:   1_analysis/setUp.Rmd
    Modified:   2_plots/1_setup/counts_before_and_after_filtering.svg
    Modified:   2_plots/1_setup/library_size.svg
    Modified:   2_plots/1_setup/sample_group_PCA_plot.svg
    Modified:   2_plots/1_setup/sample_type_PCA_plot.svg
    Deleted:    2_plots/2_dge/MA_plot_treat_CKI-Mac vs CKI.svg
    Deleted:    2_plots/2_dge/MA_plot_treat_CKI-Nme vs CKI.svg
    Deleted:    2_plots/2_dge/MA_plot_treat_CKI-Omt vs CKI.svg
    Deleted:    2_plots/2_dge/MA_plot_treat_CKI-Tri vs CKI.svg
    Deleted:    2_plots/2_dge/MA_plot_treat_UT vs CKI.svg
    Deleted:    2_plots/2_dge/estimate_dispersion.svg
    Modified:   2_plots/2_dge/qlf_dispersion.svg
    Deleted:    2_plots/2_dge/qlf_pValue_histogram_CKI-Mac vs CKI.svg
    Deleted:    2_plots/2_dge/qlf_pValue_histogram_CKI-Nme vs CKI.svg
    Deleted:    2_plots/2_dge/qlf_pValue_histogram_CKI-Omt vs CKI.svg
    Deleted:    2_plots/2_dge/qlf_pValue_histogram_CKI-Tri vs CKI.svg
    Deleted:    2_plots/2_dge/qlf_pValue_histogram_UT vs CKI.svg
    Deleted:    2_plots/2_dge/treat_pValue_histogram_CKI-Mac vs CKI.svg
    Deleted:    2_plots/2_dge/treat_pValue_histogram_CKI-Nme vs CKI.svg
    Deleted:    2_plots/2_dge/treat_pValue_histogram_CKI-Omt vs CKI.svg
    Deleted:    2_plots/2_dge/treat_pValue_histogram_CKI-Tri vs CKI.svg
    Deleted:    2_plots/2_dge/treat_pValue_histogram_UT vs CKI.svg
    Deleted:    2_plots/2_dge/volcano_plot_treat_CKI-Mac vs CKI.svg
    Deleted:    2_plots/2_dge/volcano_plot_treat_CKI-Nme vs CKI.svg
    Deleted:    2_plots/2_dge/volcano_plot_treat_CKI-Omt vs CKI.svg
    Deleted:    2_plots/2_dge/volcano_plot_treat_CKI-Tri vs CKI.svg
    Deleted:    2_plots/2_dge/volcano_plot_treat_UT vs CKI.svg
    Deleted:    3_output/enrichGO_treat.xlsx
    Deleted:    3_output/enrichKEGG_treat_unfiltered.xlsx
    Deleted:    3_output/qlf_filtered.xlsx
    Deleted:    3_output/treat_filtered.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>1_analysis/about.Rmd</code>) and HTML (<code>docs/about.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/01_CKI_DGE/blob/848e971e9ccbf102898eeb089076953d68a0817f/1_analysis/about.Rmd" target="_blank">848e971</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-11-08
</td>
<td>
adjust plot size, added dot plot for GO and KEGG, added heatmap for DE genes, minor cosmetic changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/848e971e9ccbf102898eeb089076953d68a0817f/docs/about.html" target="_blank">848e971</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-11-08
</td>
<td>
adjust plot size, added dot plot for GO and KEGG, added heatmap for DE genes, minor cosmetic changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/1a47b4569a1da246294ec62c8e47e2f6e2638e10/docs/about.html" target="_blank">1a47b45</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-31
</td>
<td>
changed heatmap size
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/70049b22b9a79841a2368454293f8825f08980dd/docs/about.html" target="_blank">70049b2</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-31
</td>
<td>
Remove old GO and KEGG code, adjusted directory paths, updated KEGG, minor cosmetic changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/6823b3c881eddd07176bce86153e0543293d7f02/docs/about.html" target="_blank">6823b3c</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
Add minor cosmetic changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/7a6571ba908cf21e984811a75b8123b1e28a4d69/docs/about.html" target="_blank">7a6571b</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
minor changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/f5f4f74bfd77b68627ec8414b040c66a60ed7344/docs/about.html" target="_blank">f5f4f74</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
please please please
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/a6d8f8a05c71e8da432f180ed252874e35ca0ae1/docs/about.html" target="_blank">a6d8f8a</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
what is happening
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/01_CKI_DGE/blob/d769a7cbf44fe4841bdf220ed214b5f73c00b47a/1_analysis/about.Rmd" target="_blank">d769a7c</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
Upload GO and KEGG analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/d769a7cbf44fe4841bdf220ed214b5f73c00b47a/docs/about.html" target="_blank">d769a7c</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
Upload GO and KEGG analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This differential gene expression analysis was performed by Ha Tran, as apart of placement with Professor David Adelson.</p>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
