<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ha Tran" />


<title>Set Up for DE Analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CKI_DGE</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Setting Up
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="obtainData.html">Obtaining Data</a>
    </li>
    <li>
      <a href="setUp.html">QC and Pre-processing</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge.html">DE Analysis</a>
    </li>
    <li>
      <a href="go.html">GO Term Enrichment Analysis</a>
    </li>
    <li>
      <a href="kegg.html">KEGG Pathway Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Set Up for DE Analysis</h1>
<h4 class="author">Ha Tran</h4>
<h4 class="date">20/08/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>01_CKI_DGE/1_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
comparison_group
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.1 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
contrast
</td>
<td style="text-align:left;">
matrix;array
</td>
<td style="text-align:left;">
1.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
designMatrix
</td>
<td style="text-align:left;">
matrix;array
</td>
<td style="text-align:left;">
3.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
dge
</td>
<td style="text-align:left;">
DGEList
</td>
<td style="text-align:left;">
12 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
enrichGO_treat
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
742 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
enrichKEGG_treat
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
491.5 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
enrichKEGG_treat_sig
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
89.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
entrezID
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
8.7 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
estimateDispersion
</td>
<td style="text-align:left;">
DGEList
</td>
<td style="text-align:left;">
12.5 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
fit_GLM_qlf
</td>
<td style="text-align:left;">
DGEGLM
</td>
<td style="text-align:left;">
20.6 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
GOresults
</td>
<td style="text-align:left;">
enrichResult
</td>
<td style="text-align:left;">
123.8 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
goSummaries
</td>
<td style="text-align:left;">
tbl_df;tbl;data.frame
</td>
<td style="text-align:left;">
13.4 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
i
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
kegg_id
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
120 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
kegg_pathway
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
51.2 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
kegg_pathway_name
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
152 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
KEGGresults
</td>
<td style="text-align:left;">
enrichResult
</td>
<td style="text-align:left;">
2.3 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
match
</td>
<td style="text-align:left;">
logical
</td>
<td style="text-align:left;">
912 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
matrix
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
13 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
matrix_merged
</td>
<td style="text-align:left;">
data.frame
</td>
<td style="text-align:left;">
1.8 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
minPath
</td>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
56 bytes
</td>
</tr>
<tr>
<td style="text-align:left;">
my_palette
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
12.6 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
qlf
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
101 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
qlf_decideTest
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
5.6 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
qlf_filtered
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
37.5 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
qlf_unfiltered
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
49.2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
treat
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
96.1 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
treat_decideTest
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
5.6 Kb
</td>
</tr>
<tr>
<td style="text-align:left;">
treat_filtered
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
1.7 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
treat_unfiltered
</td>
<td style="text-align:left;">
list
</td>
<td style="text-align:left;">
49.2 Mb
</td>
</tr>
<tr>
<td style="text-align:left;">
x
</td>
<td style="text-align:left;">
character
</td>
<td style="text-align:left;">
120 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha13501CKIDGEtreef94a382136e6b166de00c046a8ac9df415ff15e0targetblankf94a382a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tranmanhha135/01_CKI_DGE/tree/f94a382136e6b166de00c046a8ac9df415ff15e0" target="_blank">f94a382</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtranmanhha13501CKIDGEtreef94a382136e6b166de00c046a8ac9df415ff15e0targetblankf94a382a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tranmanhha135/01_CKI_DGE/tree/f94a382136e6b166de00c046a8ac9df415ff15e0" target="_blank">f94a382</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    2_plots/3_go/
    Ignored:    2_plots/4_kegg/

Untracked files:
    Untracked:  1_analysis/about.html
    Untracked:  1_analysis/dge.html
    Untracked:  1_analysis/go.html
    Untracked:  1_analysis/kegg.html
    Untracked:  1_analysis/obtainData.html
    Untracked:  1_analysis/site_libs/

Unstaged changes:
    Modified:   1_analysis/_site.yml
    Modified:   1_analysis/index.html
    Modified:   1_analysis/setUp.Rmd
    Deleted:    1_analysis/setUp.html
    Modified:   2_plots/2_dge/estimate_dispersion.svg
    Modified:   2_plots/2_dge/qlf_dispersion.svg
    Modified:   2_plots/2_dge/qlf_pValue_histogram_CKI-Mac vs CKI.svg
    Modified:   2_plots/2_dge/qlf_pValue_histogram_CKI-Nme vs CKI.svg
    Modified:   2_plots/2_dge/qlf_pValue_histogram_CKI-Omt vs CKI.svg
    Modified:   2_plots/2_dge/qlf_pValue_histogram_CKI-Tri vs CKI.svg
    Modified:   2_plots/2_dge/qlf_pValue_histogram_UT vs CKI.svg
    Modified:   2_plots/2_dge/treat_pValue_histogram_CKI-Mac vs CKI.svg
    Modified:   2_plots/2_dge/treat_pValue_histogram_CKI-Nme vs CKI.svg
    Modified:   2_plots/2_dge/treat_pValue_histogram_CKI-Omt vs CKI.svg
    Modified:   2_plots/2_dge/treat_pValue_histogram_CKI-Tri vs CKI.svg
    Modified:   2_plots/2_dge/treat_pValue_histogram_UT vs CKI.svg
    Modified:   3_output/enrichGO_treat.xlsx
    Modified:   3_output/enrichKEGG_treat_unfiltered.xlsx
    Modified:   3_output/qlf_filtered.xlsx
    Modified:   3_output/treat_filtered.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>1_analysis/setUp.Rmd</code>) and HTML (<code>docs/setUp.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/f94a382136e6b166de00c046a8ac9df415ff15e0/docs/setUp.html" target="_blank">f94a382</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
rerun website building to no avail. will it still work on github?
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/01_CKI_DGE/blob/a6d8f8a05c71e8da432f180ed252874e35ca0ae1/1_analysis/setUp.Rmd" target="_blank">a6d8f8a</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
what is happening
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/a6d8f8a05c71e8da432f180ed252874e35ca0ae1/docs/setUp.html" target="_blank">a6d8f8a</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
what is happening
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/01_CKI_DGE/blob/d769a7cbf44fe4841bdf220ed214b5f73c00b47a/1_analysis/setUp.Rmd" target="_blank">d769a7c</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
Upload GO and KEGG analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/d769a7cbf44fe4841bdf220ed214b5f73c00b47a/docs/setUp.html" target="_blank">d769a7c</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-30
</td>
<td>
Upload GO and KEGG analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tranmanhha135/01_CKI_DGE/blob/67e2d670268223ebb4a53e199040aa1830496a7f/1_analysis/setUp.Rmd" target="_blank">67e2d67</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-29
</td>
<td>
copied code from old workflow for setup and dge
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tranmanhha135/01_CKI_DGE/67e2d670268223ebb4a53e199040aa1830496a7f/docs/setUp.html" target="_blank">67e2d67</a>
</td>
<td>
Ha Tran
</td>
<td>
2021-10-29
</td>
<td>
copied code from old workflow for setup and dge
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-setup" class="section level1">
<h1>Data Setup</h1>
<div id="load-library" class="section level3">
<h3>Load Library</h3>
<pre class="r"><code># working with data
library(dplyr)
library(magrittr)
library(readr)
library(tibble)
library(reshape2)
library(tidyverse)

# Visualisation:
library(kableExtra)
library(ggbiplot)
library(ggrepel)
library(grid)
library(cowplot)

# Bioconductor packages:
library(AnnotationHub)
library(edgeR)
library(limma)
library(Glimma)
library(enrichplot)

# Set ggplot theme
theme_set(theme_bw())

theme_update(legend.background = element_rect(fill = &quot;transparent&quot;, colour = NA),
      legend.box.background = element_rect(fill = &quot;transparent&quot;, colour = NA),
      
      panel.background = element_rect(fill = &quot;transparent&quot;,colour = NA),
      plot.background = element_rect(fill = &quot;transparent&quot;, colour = NA),
      
      plot.title = element_text(color = &quot;gray20&quot;, size = 22, angle = 0, hjust = 0, vjust = .5, face = &quot;bold&quot;),
      axis.text.x = element_text(color = &quot;gray30&quot;, size = 12, angle = 0, hjust = .5, vjust = .5, face = &quot;plain&quot;),
      axis.text.y = element_text(color = &quot;gray30&quot;, size = 12, angle = 0, hjust = 1, vjust = 0, face = &quot;plain&quot;),  
      axis.title.x = element_text(color = &quot;gray30&quot;, size = 12, angle = 0, hjust = .5, vjust = 0, face = &quot;plain&quot;),
      axis.title.y = element_text(color = &quot;gray30&quot;, size = 12, angle = 90, hjust = .5, vjust = .5, face = &quot;plain&quot;))</code></pre>
</div>
<div id="data-import" class="section level2">
<h2>Data Import</h2>
<div id="import-raw-count-data" class="section level3">
<h3>Import Raw Count Data</h3>
<pre class="r"><code># raw count data was altered prior to import into R. alterations made were removing number column, removing unnecessary columns, and substituting ws delim to tab delim
rawCount &lt;- read_tsv(&quot;../0_data/raw_data/GSE125743_raw_count.tsv&quot;,
                       col_names = TRUE) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  as.data.frame</code></pre>
</div>
</div>
<div id="metadata" class="section level2">
<h2>Metadata</h2>
<p>There are generally two metadata required for DGE analysis.</p>
<ol style="list-style-type: decimal">
<li><p>metadata about <em>each sample</em></p></li>
<li><p>metadata about <em>each gene</em></p></li>
</ol>
<div id="sample-metadata" class="section level3">
<h3>Sample Metadata</h3>
<p>The sample metadata can be extracted from the <code>GSE125743_raw_count.tsv</code> column names. These data include <code>sample_id</code>, <code>sample_group</code>, <code>sample_type</code>.</p>
<p><strong>Current Challenges</strong></p>
<p><strong>1. using obtaining <code>sample_group</code>, <code>sample_type</code> through regex</strong> <strong>2. using conditional to add suffix of the coloumns</strong></p>
<p>Due to these unresolved issues, the sample metadata will be manually generated and stored in the <code>CKI_DGE/0_data/raw_data/</code> directory</p>
<pre class="r"><code>sampleMetadata &lt;- read_tsv(&quot;../0_data/raw_data/sample_metadata.tsv&quot;,
                           col_names = TRUE) %&gt;% 
  column_to_rownames(&quot;treatment&quot;) %&gt;% 
  as.data.frame</code></pre>
</div>
<div id="gene-metadata" class="section level3">
<h3>Gene Metadata</h3>
<ul>
<li>Gene annotation is useful for the latter DGE analysis as it will provide useful information about the genes. The annotated genes of <em>Homo sapiens</em> can be pulled down by using Annotation Hub.</li>
<li>Annotation Hub also has a web service that can be assessed through the display function. Pulling down the gene annotation can take a long time, so after the initial run, the annotated genes is saved to a <code>genes.rds</code> file. To save time, if <code>genes.rds</code> is already present, don’t run the code chunk.</li>
</ul>
<pre class="r"><code>ah &lt;- AnnotationHub()
ah %&gt;% 
  subset(grepl(&quot;sapiens&quot;, species)) %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;)

#viewing web service for annotation hub 
#d &lt;- display(ah)

#Annotation hub html site was used to identify &#39;code&#39; for the latest human  genome from Ensembl
ensDb &lt;- ah[[&quot;AH95744&quot;]]

genes &lt;- genes(ensDb) %&gt;% 
  as.data.frame()

#the annotated genes are saved into a RDS object to save computational time in subsequent run of the setUp.Rmd
genes %&gt;% saveRDS(&quot;../0_data/rds_object/genes_metadata.rds&quot;)</code></pre>
<p>Using the annotated gene list through AnnotationHub(), load into object called <code>geneMetadata</code>. Filter out all genes that are present in the rawCount and display the number of unique gene_biotypes present in the rawCount and geneMetadata</p>
<pre class="r"><code>genesMetadata &lt;- read_rds(&quot;../0_data/rds_object/genes_metadata.rds&quot;)

#prepare the gene data frame to contain the genes listed in the rownames of &#39;rawCount&#39; data
genesMetadata &lt;- data.frame(gene = rownames(rawCount)) %&gt;% 
  left_join(genesMetadata %&gt;% as.data.frame, 
            by = c(&quot;gene&quot;=&quot;symbol&quot;)) %&gt;%
  dplyr::distinct(gene, .keep_all=TRUE)

#make the gene column the row names
rownames(genesMetadata) &lt;- genesMetadata$gene

#Using the table function, the details of the genes present in the rawCount data can be summaried.
genes &lt;- genesMetadata$gene_biotype %&gt;% table %&gt;% as.data.frame()

kable(genes) %&gt;% kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;)) %&gt;% scroll_box(height = &quot;600px&quot;)</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:600px; ">
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
.
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IG_V_gene
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
IG_V_pseudogene
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
lncRNA
</td>
<td style="text-align:right;">
3261
</td>
</tr>
<tr>
<td style="text-align:left;">
LRG_gene
</td>
<td style="text-align:right;">
175
</td>
</tr>
<tr>
<td style="text-align:left;">
miRNA
</td>
<td style="text-align:right;">
1839
</td>
</tr>
<tr>
<td style="text-align:left;">
misc_RNA
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
polymorphic_pseudogene
</td>
<td style="text-align:right;">
32
</td>
</tr>
<tr>
<td style="text-align:left;">
processed_pseudogene
</td>
<td style="text-align:right;">
144
</td>
</tr>
<tr>
<td style="text-align:left;">
protein_coding
</td>
<td style="text-align:right;">
18642
</td>
</tr>
<tr>
<td style="text-align:left;">
ribozyme
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
rRNA
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
scaRNA
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
scRNA
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
snoRNA
</td>
<td style="text-align:right;">
419
</td>
</tr>
<tr>
<td style="text-align:left;">
snRNA
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
TEC
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
transcribed_processed_pseudogene
</td>
<td style="text-align:right;">
100
</td>
</tr>
<tr>
<td style="text-align:left;">
transcribed_unitary_pseudogene
</td>
<td style="text-align:right;">
79
</td>
</tr>
<tr>
<td style="text-align:left;">
transcribed_unprocessed_pseudogene
</td>
<td style="text-align:right;">
406
</td>
</tr>
<tr>
<td style="text-align:left;">
translated_unprocessed_pseudogene
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
unitary_pseudogene
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
unprocessed_pseudogene
</td>
<td style="text-align:right;">
117
</td>
</tr>
<tr>
<td style="text-align:left;">
vault_RNA
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="create-dgelist-object" class="section level2">
<h2>Create DGEList object</h2>
<p>Digital Gene Expression List (DGElist) is a R object class often used for differential gene expression analysis as it simplifies plotting, and interaction with data and metadata.</p>
<p>The DGEList object holds the three dataset that have imported/created, including <code>rawCount</code> data and <code>sampleMetadata</code> and <code>geneMetadata</code> metadata.</p>
<p>To further save time and memory, genes that were not expressed acrossed all samples (i.e., <code>0</code> count across all columns) are all removed</p>
<pre class="r"><code>#Create DGElist with rawCOunt and gene data. Remove all genes with 0 expression in all treatment groups
dge &lt;- edgeR::DGEList(
  counts = rawCount,
  samples = sampleMetadata,
  genes = genesMetadata,
  remove.zeros = TRUE)</code></pre>
<p>5207 non expressed genes (rows) were removed.</p>
</div>
</div>
<div id="pre-processing-and-quality-checks" class="section level1">
<h1>Pre-processing and Quality Checks</h1>
<p>Pre-processing steps increased the power of the downstream DGE analysis by eliminating majority of unwanted variance that could obscure the true variance caused by the differences in sample conditions. There are several standard steps that are commonly followed to pre-process and QC raw read counts, including:</p>
<ol style="list-style-type: decimal">
<li><p>Checking Library Size</p></li>
<li><p>Removal of Undetectable Genes</p></li>
<li><p>Normalisation</p></li>
<li><p>QC through MDS/PCA</p></li>
</ol>
<div id="checking-library-size" class="section level2">
<h2>Checking Library Size</h2>
<p>A simple pre-processing/QC step is checking the quality of library size (total number of mapped and quantified reads) for each treatment. This enable identification of potentially mis-labeled or outlying samples. This is often visualised through <code>ggplot</code>.</p>
<pre class="r"><code>dge$samples %&gt;% 
  
  #plot the sample with the lib.size in x and sample_group in y, colour fill for each sample_group
  ggplot(aes(
    x = lib.size, 
    y = pretty_name, 
    fill = sample_group)
    ) +
  geom_col() +
  
  #draw a vertical line for the mean lib.size
  geom_vline(
    aes (xintercept = lib.size), 
    data = . %&gt;% summarise_at(vars(lib.size), mean),
    linetype = 2
  ) +
  

  theme(legend.position = &quot;none&quot;) +
  
  #labelling splot
  labs(title = &quot;Sample Library Size&quot;,
    x = &quot;Library Size&quot;,
    y = &quot;Samples&quot;
  )</code></pre>
<p><img src="figure/setUp.Rmd/libraryChecks-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#save the plot to .svg
ggsave(filename = &quot;library_size.svg&quot;,
       plot = last_plot(),
       path = &quot;../2_plots/1_setup/&quot;)</code></pre>
<p>It appears that <code>CKI-Tri</code> treatment group</p>
</div>
<div id="removal-of-low-expressed-genes" class="section level2">
<h2>Removal of Low-Expressed Genes</h2>
<p>Filtering out low-expressed genes is a standard pre-processing step in DGE analysis as it can significantly increase the power to differentiate differentially expressed genes by eliminating the variance caused by genes that are lowly expressed in all samples.</p>
<p>The threshold of removal is arbitrary and is often determined after visualisation of the count distribution. The count distribution can be illustrated in a density plot through <code>ggplot</code>. A common metric used to display the count distribution if <em>log Counts per Million</em> (logCPM)</p>
<pre class="r"><code>beforeFiltering &lt;- dge %&gt;% 
  
  #transform the raw count to logCPM 
  edgeR::cpm(log = TRUE) %&gt;% 

  #melting (reorganising) the transformed logCPM data with respect to the id variable (i.e., the row and column names). Very similar to pivot_long fucntion
  melt %&gt;% 
  
  #retain all rows where the logCPM (value) is finite. ALl of them in this case are finite
  dplyr::filter(is.finite(value)) %&gt;% 
  
  #plot the long formate logCPM counts in a density plot with x the logCPM and colour by the sample_id
  ggplot(aes(
    x = value,
    colour = Var2
  )) +
  geom_density() +
    
  #remove the legend
  guides(colour = FALSE) +
  
  #add figure tittle and subtitle and labels
  ggtitle(&quot;Before Filtering Low-expressed Genes&quot;, subtitle = paste0(nrow(dge), &quot; genes&quot;))+
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
    colour = &quot;Sample Groups&quot;
  )

#save plot
# ggsave(filename = &quot;counts_before_filtering.svg&quot;, 
#        plot = beforeFiltering, 
#        path = &quot;../2_plots/1_setup/&quot;)</code></pre>
<p>Ideally, the filtering the low-expressed genes should remove the large peak with <code>logCPM &lt; 0</code>, i.e., remove any genes which have less than one count per million.</p>
<p>A common guideline is to keep all genes that have &gt; 1 cpm in the smallest group on a treatment. In this case, the smallest group is 3 as each treatment condition had three replicates. Therefore any genes that is <strong>more than 1 CPM in at least 3 samples</strong> is kept. Mathematically this would be identifying genes (rows) with CPM <code>&gt; 1</code>; and identifying total row sum that is <code>&gt;= 3</code>.</p>
<pre class="r"><code>#kept genes old method
# keep &lt;- edgeR::filterByExpr(y = dge,)

#the genes kept have &gt;1 CPM for at least 3 samples
keptGenes &lt;- (rowSums(cpm(dge) &gt; 1.5) &gt;= 3) 

afterFiltering &lt;- dge %&gt;% 
  
  #transform the raw count to logCPM 
  edgeR::cpm(log = TRUE) %&gt;% 
  
  #for var1 (gene names) extract only the keptGenes and discard all other genes in the logCPM data
  magrittr::extract(keptGenes,) %&gt;% 

  #melting (reorganising) the transformed logCPM data with respect to the id variable (i.e., the row and column names). Very similar to pivot_long fucntion
  melt %&gt;% 
  
  #retain all rows where the logCPM (value) is finite. ALl of them in this case are finite
  dplyr::filter(is.finite(value)) %&gt;% 

  #ggplot
  ggplot(aes(
    x = value,
    colour = Var2
  )) +
  geom_density() +
  
  #remove the legend
  guides(colour = FALSE) +
  
  #add figure tittle and subtitle and labels. since keptGenes is a logic element, the second element represents the number of genes that were kept after the filtering
  ggtitle(&quot;After Filtering Low-expressed Genes&quot;, subtitle = paste0(table(keptGenes)[[2]], &quot; genes&quot;)) +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
    colour = &quot;Sample Groups&quot;
  )

#save plot
# ggsave(&quot;counts_after_filtering.svg&quot;, 
#        plot = afterFiltering,
#        path = &quot;../2_plots/1_setup/&quot;)


#display plot side by side
beforeAfterFiltering &lt;- cowplot::plot_grid(beforeFiltering, afterFiltering)

#save plot
ggsave(&quot;counts_before_and_after_filtering.svg&quot;,
       width = 18, 
       plot = beforeAfterFiltering,
       path = &quot;../2_plots/1_setup/&quot;)

#display side by side figure
beforeAfterFiltering</code></pre>
<p><img src="figure/setUp.Rmd/afterFiltering-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>Following the filtering of low-expressed genes <strong>&lt; 1.5 CPM in at least 3 samples</strong>, out of the total 22771 genes left after the removal of genes with no expression, 9512 genes were removed, leaving only 13259 genes remaining for the downstream analysis</p>
<div id="subset-the-dgelist-object" class="section level3">
<h3>Subset the DGElist object</h3>
<p>After filtering the low-expressed genes, the DGElist object is updated to eliminate the low-expressed genes from future analysis</p>
<pre class="r"><code>#extract genes from keptGenes and recalculate the lib size
dge &lt;- dge[keptGenes,,keep.lib.sizes = FALSE]</code></pre>
</div>
</div>
<div id="normalisation" class="section level2">
<h2>Normalisation</h2>
<p>Using the TMM (trimmed mean of M value) method of normalisation through the <code>edgeR</code> package. The TMM approach creates a scaling factor as an offset to be supplied to Negative Binomial model. The <code>calcNormFactors</code> function calculate the normalisation and return the adjusted <code>norm.factor</code> to the <code>dge$samples</code> element.</p>
<div id="before-normalisation" class="section level3">
<h3>Before Normalisation</h3>
<pre class="r"><code>#before normalisation
dge$samples</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["sample_id"],"name":[4],"type":["chr"],"align":["left"]},{"label":["sample_group"],"name":[5],"type":["chr"],"align":["left"]},{"label":["sample_type"],"name":[6],"type":["chr"],"align":["left"]},{"label":["pretty_name"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"1","2":"19131407","3":"1","4":"S48h_CKI1","5":"CKI","6":"CKI","7":"CKI 1","_rn_":"S48h_CKI1"},{"1":"1","2":"22169636","3":"1","4":"S48h_CKI2","5":"CKI","6":"CKI","7":"CKI 2","_rn_":"S48h_CKI2"},{"1":"1","2":"22201831","3":"1","4":"S48h_CKI3","5":"CKI","6":"CKI","7":"CKI 3","_rn_":"S48h_CKI3"},{"1":"1","2":"22039619","3":"1","4":"S48h_Mac1","5":"Mac","6":"single_deletion","7":"CKI-Mac 1","_rn_":"S48h_Mac1"},{"1":"1","2":"20998943","3":"1","4":"S48h_Mac2","5":"Mac","6":"single_deletion","7":"CKI-Mac 2","_rn_":"S48h_Mac2"},{"1":"1","2":"22108586","3":"1","4":"S48h_Mac3","5":"Mac","6":"single_deletion","7":"CKI-Mac 3","_rn_":"S48h_Mac3"},{"1":"1","2":"22773367","3":"1","4":"S48h_Nme1","5":"Nme","6":"single_deletion","7":"CKI-Nme 1","_rn_":"S48h_Nme1"},{"1":"1","2":"20627664","3":"1","4":"S48h_Nme2","5":"Nme","6":"single_deletion","7":"CKI-Nme 2","_rn_":"S48h_Nme2"},{"1":"1","2":"24570009","3":"1","4":"S48h_Nme3","5":"Nme","6":"single_deletion","7":"CKI-Nme 3","_rn_":"S48h_Nme3"},{"1":"1","2":"22859586","3":"1","4":"S48h_OMT1","5":"Omt","6":"single_deletion","7":"CKI-Omt 1","_rn_":"S48h_OMT1"},{"1":"1","2":"23358998","3":"1","4":"S48h_OMT2","5":"Omt","6":"single_deletion","7":"CKI-Omt 2","_rn_":"S48h_OMT2"},{"1":"1","2":"19166567","3":"1","4":"S48h_OMT3","5":"Omt","6":"single_deletion","7":"CKI-Omt 3","_rn_":"S48h_OMT3"},{"1":"1","2":"17648364","3":"1","4":"S48h_Tri1","5":"Tri","6":"single_deletion","7":"CKI-Tri 1","_rn_":"S48h_Tri1"},{"1":"1","2":"24460016","3":"1","4":"S48h_Tri2","5":"Tri","6":"single_deletion","7":"CKI-Tri 2","_rn_":"S48h_Tri2"},{"1":"1","2":"21036767","3":"1","4":"S48h_Tri3","5":"Tri","6":"single_deletion","7":"CKI-Tri 3","_rn_":"S48h_Tri3"},{"1":"1","2":"25420911","3":"1","4":"S48h_UT1","5":"UT","6":"UT","7":"Untreated 1","_rn_":"S48h_UT1"},{"1":"1","2":"23342210","3":"1","4":"S48h_UT2","5":"UT","6":"UT","7":"Untreated 2","_rn_":"S48h_UT2"},{"1":"1","2":"24238281","3":"1","4":"S48h_UT3","5":"UT","6":"UT","7":"Untreated 3","_rn_":"S48h_UT3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="after-normalisation" class="section level3">
<h3>After Normalisation</h3>
<pre class="r"><code>#after normalisation
dge &lt;- edgeR::calcNormFactors(object = dge, 
                              method = &quot;TMM&quot;)
dge$samples</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["sample_id"],"name":[4],"type":["chr"],"align":["left"]},{"label":["sample_group"],"name":[5],"type":["chr"],"align":["left"]},{"label":["sample_type"],"name":[6],"type":["chr"],"align":["left"]},{"label":["pretty_name"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"1","2":"19131407","3":"1.0264826","4":"S48h_CKI1","5":"CKI","6":"CKI","7":"CKI 1","_rn_":"S48h_CKI1"},{"1":"1","2":"22169636","3":"1.0312022","4":"S48h_CKI2","5":"CKI","6":"CKI","7":"CKI 2","_rn_":"S48h_CKI2"},{"1":"1","2":"22201831","3":"1.0252916","4":"S48h_CKI3","5":"CKI","6":"CKI","7":"CKI 3","_rn_":"S48h_CKI3"},{"1":"1","2":"22039619","3":"1.0006686","4":"S48h_Mac1","5":"Mac","6":"single_deletion","7":"CKI-Mac 1","_rn_":"S48h_Mac1"},{"1":"1","2":"20998943","3":"1.0115535","4":"S48h_Mac2","5":"Mac","6":"single_deletion","7":"CKI-Mac 2","_rn_":"S48h_Mac2"},{"1":"1","2":"22108586","3":"1.0076475","4":"S48h_Mac3","5":"Mac","6":"single_deletion","7":"CKI-Mac 3","_rn_":"S48h_Mac3"},{"1":"1","2":"22773367","3":"0.9892032","4":"S48h_Nme1","5":"Nme","6":"single_deletion","7":"CKI-Nme 1","_rn_":"S48h_Nme1"},{"1":"1","2":"20627664","3":"1.0114780","4":"S48h_Nme2","5":"Nme","6":"single_deletion","7":"CKI-Nme 2","_rn_":"S48h_Nme2"},{"1":"1","2":"24570009","3":"1.0240423","4":"S48h_Nme3","5":"Nme","6":"single_deletion","7":"CKI-Nme 3","_rn_":"S48h_Nme3"},{"1":"1","2":"22859586","3":"0.9733067","4":"S48h_OMT1","5":"Omt","6":"single_deletion","7":"CKI-Omt 1","_rn_":"S48h_OMT1"},{"1":"1","2":"23358998","3":"0.9762419","4":"S48h_OMT2","5":"Omt","6":"single_deletion","7":"CKI-Omt 2","_rn_":"S48h_OMT2"},{"1":"1","2":"19166567","3":"0.9746798","4":"S48h_OMT3","5":"Omt","6":"single_deletion","7":"CKI-Omt 3","_rn_":"S48h_OMT3"},{"1":"1","2":"17648364","3":"1.0212643","4":"S48h_Tri1","5":"Tri","6":"single_deletion","7":"CKI-Tri 1","_rn_":"S48h_Tri1"},{"1":"1","2":"24460016","3":"1.0242485","4":"S48h_Tri2","5":"Tri","6":"single_deletion","7":"CKI-Tri 2","_rn_":"S48h_Tri2"},{"1":"1","2":"21036767","3":"1.0205099","4":"S48h_Tri3","5":"Tri","6":"single_deletion","7":"CKI-Tri 3","_rn_":"S48h_Tri3"},{"1":"1","2":"25420911","3":"0.9614313","4":"S48h_UT1","5":"UT","6":"UT","7":"Untreated 1","_rn_":"S48h_UT1"},{"1":"1","2":"23342210","3":"0.9653451","4":"S48h_UT2","5":"UT","6":"UT","7":"Untreated 2","_rn_":"S48h_UT2"},{"1":"1","2":"24238281","3":"0.9609013","4":"S48h_UT3","5":"UT","6":"UT","7":"Untreated 3","_rn_":"S48h_UT3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="visualisation-of-tmm-normalisation" class="section level3">
<h3>Visualisation of TMM Normalisation</h3>
<p>The following visualisation of the TMM normalisation is plotted using the mean-difference (MD) plot. The MD plot visualise the library size-adjusted logFC between two samples (the difference) against the log-expression across all samples (the mean). In this instance, <code>sample 1</code> is used to compare against an artificial library construct from the average of all the other samples</p>
<pre class="r"><code>limma::plotMD(cpm(dge, log = TRUE), column=1)
abline(h=0, col=&quot;red&quot;, lty=2, lwd=2)</code></pre>
<p><img src="figure/setUp.Rmd/visualisingNormalisation-1.png" width="1056" style="display: block; margin: auto;" /> Ideally, the bulk of gene expression following the TMM normalisation should be centred around <code>expression log-ratio</code> of 0, which indicates that library size bias between samples have been successfully removed. This should be repeated with all the samples in the dge object.</p>
</div>
</div>
<div id="quality-control" class="section level2">
<h2>Quality Control</h2>
<div id="multi-dimensional-scaling-mds" class="section level3">
<h3>Multi Dimensional Scaling (MDS)</h3>
<p>Multiple dimension reduction method to assess clustering of samples</p>
<pre class="r"><code># #interactive MDS
# glMDSPlot(dge,
#           labels = dge$samples$sample,
#           groups = dge$samples$sample_group,
#           folder = &quot;../5_plots/1_QC_plots/&quot;)

#non-interactive MDS plot
mdsPlot &lt;-  limma::plotMDS(dge,
                            main = &quot;MDS Plot for Count Data&quot;,
                            top = 500,
                            labels = colnames(dge$counts),
                            col = as.numeric(dge$samples$sample_id),
                            plot = TRUE)</code></pre>
<p><img src="figure/setUp.Rmd/MDSplot-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="pinciple-component-analysis-pca" class="section level3">
<h3>Pinciple Component Analysis (PCA)</h3>
<div id="pca-analysis" class="section level4">
<h4>PCA Analysis</h4>
<pre class="r"><code># Perform PCA analysis:
pcaAnalysis &lt;- prcomp(t(cpm(dge, log=TRUE)))
summary(pcaAnalysis)$importance %&gt;% as.data.frame()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["PC1"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["PC2"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["PC3"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["PC4"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["PC5"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["PC6"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["PC7"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["PC8"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["PC9"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["PC10"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["PC11"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["PC12"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["PC13"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["PC14"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["PC15"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["PC16"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["PC17"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["PC18"],"name":[18],"type":["dbl"],"align":["right"]}],"data":[{"1":"40.40065","2":"23.76823","3":"19.97026","4":"10.06435","5":"6.658135","6":"4.80615","7":"4.130353","8":"4.046085","9":"3.943185","10":"3.873321","11":"3.814304","12":"3.710899","13":"3.656684","14":"3.618051","15":"3.558747","16":"3.477436","17":"3.44890","18":"4.506029e-14","_rn_":"Standard deviation"},{"1":"0.55896","2":"0.19346","3":"0.13658","4":"0.03469","5":"0.015180","6":"0.00791","7":"0.005840","8":"0.005610","9":"0.005320","10":"0.005140","11":"0.004980","12":"0.004720","13":"0.004580","14":"0.004480","15":"0.004340","16":"0.004140","17":"0.00407","18":"0.000000e+00","_rn_":"Proportion of Variance"},{"1":"0.55896","2":"0.75242","3":"0.88900","4":"0.92369","5":"0.938870","6":"0.94678","7":"0.952620","8":"0.958230","9":"0.963550","10":"0.968690","11":"0.973670","12":"0.978390","13":"0.982970","14":"0.987450","15":"0.991790","16":"0.995930","17":"1.00000","18":"1.000000e+00","_rn_":"Cumulative Proportion"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="pca-of-sample-type" class="section level4">
<h4>PCA of Sample Type</h4>
<pre class="r"><code># Create the plot:
pcaSampleType &lt;- ggbiplot(pcaAnalysis,
                              groups = dge$samples$sample_type,
                              ellipse = TRUE,
                              var.axes = FALSE) +
  labs(title = &quot;PCA of CKI Sample Type&quot;) +
  theme(legend.position = &quot;bottom&quot;)

ggsave(filename = &quot;sample_type_PCA_plot.svg&quot;, width = 20,
       plot = pcaSampleType,
       path = &quot;../2_plots/1_setup/&quot;)

pcaSampleType</code></pre>
<p><img src="figure/setUp.Rmd/pcaSampleType-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-of-sample-group" class="section level4">
<h4>PCA of Sample Group</h4>
<pre class="r"><code># Create the plot:
pcaSampleGroup &lt;- ggbiplot(pcaAnalysis,
                           groups = dge$samples$sample_group,
                           ellipse = TRUE,
                           var.axes = FALSE) +
  labs(title = &quot;PCA of CKI Sample Group&quot;) +
  theme(legend.position = &quot;bottom&quot;)
        

ggsave(filename = &quot;sample_group_PCA_plot.svg&quot;, width = 20,
       plot = pcaSampleGroup,
       path = &quot;../2_plots/1_setup/&quot;)

pcaSampleGroup</code></pre>
<p><img src="figure/setUp.Rmd/pcaSampleGroup-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>The clustering of sample replicates and separation of sample groups indicates that the differences between the groups are greater than the variance within the groups. Thus, the differential gene expression is greater than the variance</p>
</div>
</div>
</div>
<div id="save-dgelist-object" class="section level2">
<h2>Save DGElist object</h2>
<pre class="r"><code>saveRDS(object = dge, file = &quot;../0_data/rds_object/dge.rds&quot;)</code></pre>
</div>
</div>
<div id="r-workflowrwflow_build" class="section level1">
<h1><code>{r} # workflowr::wflow_build() #</code></h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] AnnotationHub_3.0.2   BiocFileCache_2.0.0   dbplyr_2.1.1         
 [4] enrichplot_1.12.3     KEGGREST_1.32.0       biomaRt_2.48.3       
 [7] org.Hs.eg.db_3.13.0   AnnotationDbi_1.54.1  IRanges_2.26.0       
[10] S4Vectors_0.30.0      Biobase_2.52.0        BiocGenerics_0.38.0  
[13] clusterProfiler_4.0.5 Glimma_2.2.0          edgeR_3.34.0         
[16] limma_3.48.1          ggrepel_0.9.1         ggbiplot_0.55        
[19] scales_1.1.1          plyr_1.8.6            ggpubr_0.4.0         
[22] pheatmap_1.0.12       cowplot_1.1.1         pander_0.6.4         
[25] kableExtra_1.3.4      forcats_0.5.1         stringr_1.4.0        
[28] purrr_0.3.4           tidyr_1.1.3           ggplot2_3.3.5        
[31] tidyverse_1.3.1       reshape2_1.4.4        tibble_3.1.2         
[34] readr_2.0.0           magrittr_2.0.1        dplyr_1.0.7          

loaded via a namespace (and not attached):
  [1] utf8_1.2.1                    tidyselect_1.1.1             
  [3] RSQLite_2.2.7                 htmlwidgets_1.5.4            
  [5] BiocParallel_1.26.1           scatterpie_0.1.7             
  [7] munsell_0.5.0                 statmod_1.4.36               
  [9] withr_2.4.2                   colorspace_2.0-2             
 [11] GOSemSim_2.18.1               filelock_1.0.2               
 [13] highr_0.9                     knitr_1.36                   
 [15] rstudioapi_0.13               ggsignif_0.6.3               
 [17] DOSE_3.18.3                   MatrixGenerics_1.4.3         
 [19] labeling_0.4.2                git2r_0.28.0                 
 [21] GenomeInfoDbData_1.2.6        polyclip_1.10-0              
 [23] bit64_4.0.5                   farver_2.1.0                 
 [25] rprojroot_2.0.2               downloader_0.4               
 [27] treeio_1.16.2                 vctrs_0.3.8                  
 [29] generics_0.1.0                xfun_0.24                    
 [31] R6_2.5.1                      GenomeInfoDb_1.28.4          
 [33] graphlayouts_0.7.1            locfit_1.5-9.4               
 [35] gridGraphics_0.5-1            bitops_1.0-7                 
 [37] cachem_1.0.5                  fgsea_1.18.0                 
 [39] DelayedArray_0.18.0           assertthat_0.2.1             
 [41] vroom_1.5.5                   promises_1.2.0.1             
 [43] ggraph_2.0.5                  gtable_0.3.0                 
 [45] tidygraph_1.2.0               workflowr_1.6.2              
 [47] rlang_0.4.11                  genefilter_1.74.0            
 [49] systemfonts_1.0.2             splines_4.1.0                
 [51] lazyeval_0.2.2                rstatix_0.7.0                
 [53] broom_0.7.9                   BiocManager_1.30.16          
 [55] yaml_2.2.1                    abind_1.4-5                  
 [57] modelr_0.1.8                  backports_1.2.1              
 [59] httpuv_1.6.1                  qvalue_2.24.0                
 [61] tools_4.1.0                   ggplotify_0.1.0              
 [63] ellipsis_0.3.2                jquerylib_0.1.4              
 [65] RColorBrewer_1.1-2            Rcpp_1.0.6                   
 [67] progress_1.2.2                zlibbioc_1.38.0              
 [69] RCurl_1.98-1.3                prettyunits_1.1.1            
 [71] viridis_0.6.2                 SummarizedExperiment_1.22.0  
 [73] haven_2.4.3                   fs_1.5.0                     
 [75] data.table_1.14.0             DO.db_2.9                    
 [77] openxlsx_4.2.4                reprex_2.0.1                 
 [79] whisker_0.4                   matrixStats_0.60.1           
 [81] mime_0.11                     patchwork_1.1.1              
 [83] hms_1.1.1                     evaluate_0.14                
 [85] xtable_1.8-4                  XML_3.99-0.6                 
 [87] rio_0.5.27                    readxl_1.3.1                 
 [89] gridExtra_2.3                 compiler_4.1.0               
 [91] shadowtext_0.0.9              writexl_1.4.0                
 [93] crayon_1.4.1                  htmltools_0.5.2              
 [95] ggfun_0.0.4                   later_1.2.0                  
 [97] tzdb_0.1.2                    aplot_0.1.1                  
 [99] geneplotter_1.70.0            lubridate_1.7.10             
[101] DBI_1.1.1                     tweenr_1.0.2                 
[103] rappdirs_0.3.3                MASS_7.3-54                  
[105] Matrix_1.3-3                  car_3.0-11                   
[107] cli_3.0.1                     igraph_1.2.6                 
[109] GenomicRanges_1.44.0          pkgconfig_2.0.3              
[111] foreign_0.8-81                xml2_1.3.2                   
[113] ggtree_3.0.4                  svglite_2.0.0                
[115] annotate_1.70.0               bslib_0.3.1                  
[117] webshot_0.5.2                 XVector_0.32.0               
[119] rvest_1.0.2                   yulab.utils_0.0.4            
[121] digest_0.6.27                 Biostrings_2.60.1            
[123] rmarkdown_2.11                cellranger_1.1.0             
[125] fastmatch_1.1-3               tidytree_0.3.5               
[127] curl_4.3.2                    shiny_1.7.1                  
[129] lifecycle_1.0.1               nlme_3.1-152                 
[131] jsonlite_1.7.2                carData_3.0-4                
[133] viridisLite_0.4.0             fansi_0.5.0                  
[135] pillar_1.6.4                  lattice_0.20-44              
[137] fastmap_1.1.0                 httr_1.4.2                   
[139] survival_3.2-11               GO.db_3.13.0                 
[141] interactiveDisplayBase_1.30.0 glue_1.4.2                   
[143] zip_2.2.0                     png_0.1-7                    
[145] BiocVersion_3.13.1            bit_4.0.4                    
[147] ggforce_0.3.3                 stringi_1.6.2                
[149] sass_0.4.0                    blob_1.2.2                   
[151] DESeq2_1.32.0                 memoise_2.0.0                
[153] ape_5.5                      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
